src/Aspire.Hosting.AWS/CloudFormation/CloudFormationExtensions.cs

```diff
@@ -0,0 +1,193 @@
+        {
+            if (ctx.ExecutionContext.IsPublishMode)
+            {
+                ctx.EnvironmentVariables[name] = stackOutputReference.ValueExpression;
+                return;
+            }
+
+            ctx.Logger?.LogInformation("Getting CloudFormation stack output {Name} from resource {ResourceName}", stackOutputReference.Name, stackOutputReference.Resource.Name);
+
+            ctx.EnvironmentVariables[name] = await stackOutputReference.GetValueAsync(ctx.CancellationToken).ConfigureAwait(false) ?? "";
+        });
+    }
+
+    /// <summary>
+    /// The AWS SDK service client configuration used to create the CloudFormation service client.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <param name="awsSdkConfig">The name of the AWS credential profile.</param>
+    public static IResourceBuilder<ICloudFormationResource> WithReference(this IResourceBuilder<ICloudFormationResource> builder, IAWSSDKConfig awsSdkConfig)
+    {
+        builder.Resource.AWSSDKConfig = awsSdkConfig;
+        return builder;
+    }
+
+    /// <summary>
+    /// Override the CloudFormation service client the ICloudFormationResource would create to interact with the CloudFormation service. This can be used for pointing the
+    /// CloudFormation service client to a non-standard CloudFormation endpoint like an emulator.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <param name="cloudFormationClient">The AWS CloudFormation service client.</param>
+    public static IResourceBuilder<ICloudFormationResource> WithReference(this IResourceBuilder<ICloudFormationResource> builder, IAmazonCloudFormation cloudFormationClient)
+    {
+        builder.Resource.CloudFormationClient = cloudFormationClient;
+        return builder;
+    }
+
+    /// <summary>
+    /// Add a reference of a CloudFormations stack to a project. The output parameters of the CloudFormation stack are added to the project IConfiguration.
+    /// </summary>
+    /// <typeparam name="TDestination"></typeparam>
+    /// <param name="builder"></param>
+    /// <param name="cloudFormationResourceBuilder">The CloudFormation resource.</param>
+    /// <param name="configSection">The config section in IConfiguration to add the output parameters.</param>
+    /// <returns></returns>
+    public static IResourceBuilder<TDestination> WithReference<TDestination>(this IResourceBuilder<TDestination> builder, IResourceBuilder<ICloudFormationResource> cloudFormationResourceBuilder, string configSection = "AWS::Resources")
+        where TDestination : IResourceWithEnvironment
+    {
+        var referenceResource = builder.ApplicationBuilder.AddResource(new CloudFormationReferenceResource(cloudFormationResourceBuilder.Resource, builder.Resource));
+        referenceResource.WithManifestPublishingCallback(context => WriteCloudFormationReference(context, cloudFormationResourceBuilder.Resource, builder.Resource));
```

Comment:
Why is this needed?

```suggestion
```

Reply:
Just model these as annotations, not resources. Or just data on the resource itself?

Reply:
Replaced with the new `CloudFormationReferenceAnnotation` annotation that the manifest writer for the CloudFormation resources uses to add the references to their manifest resource.

```
{
  "resources": {
    "AspireSampleDevResources": {
      "type": "aws.cloudformation.template.v0",
      "stack-name": "AspireSampleDevResources",
      "template-path": "app-resources.template",
      "references": [
        {
          "TargetResource": "Frontend"
        }
      ]
    },
    "Frontend": {
      "type": "project.v0",
...

```